plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/AppEngineProjectWizard.java

```diff
@@ -20,52 +20,51 @@
 import com.google.cloud.tools.appengine.cloudsdk.CloudSdk;
 import com.google.cloud.tools.appengine.cloudsdk.CloudSdkNotFoundException;
 import com.google.cloud.tools.appengine.cloudsdk.CloudSdkOutOfDateException;
-import com.google.cloud.tools.eclipse.appengine.libraries.ILibraryClasspathContainerResolverService;
-import com.google.cloud.tools.eclipse.appengine.libraries.ILibraryClasspathContainerResolverService.AppEngineRuntime;
 import com.google.cloud.tools.eclipse.appengine.ui.AppEngineJavaComponentMissingPage;
 import com.google.cloud.tools.eclipse.appengine.ui.CloudSdkMissingPage;
 import com.google.cloud.tools.eclipse.appengine.ui.CloudSdkOutOfDatePage;
 import com.google.cloud.tools.eclipse.sdk.ui.preferences.CloudSdkPrompter;
 import com.google.cloud.tools.eclipse.ui.util.WorkbenchUtil;
 import com.google.cloud.tools.eclipse.usagetracker.AnalyticsEvents;
-import com.google.cloud.tools.eclipse.usagetracker.AnalyticsPingManager;
 import com.google.cloud.tools.eclipse.util.status.StatusUtil;
 import java.io.File;
 import java.lang.reflect.InvocationTargetException;
-import javax.inject.Inject;
 import org.eclipse.core.resources.IFile;
 import org.eclipse.core.runtime.IAdaptable;
-import org.eclipse.core.runtime.IProgressMonitor;
 import org.eclipse.core.runtime.IStatus;
 import org.eclipse.core.runtime.Status;
-import org.eclipse.jface.operation.IRunnableWithProgress;
 import org.eclipse.jface.viewers.IStructuredSelection;
 import org.eclipse.jface.wizard.Wizard;
 import org.eclipse.ui.INewWizard;
 import org.eclipse.ui.IWorkbench;
 import org.eclipse.ui.ide.undo.WorkspaceUndoUtil;
 
-public class StandardProjectWizard extends Wizard implements INewWizard {
+public abstract class AppEngineProjectWizard extends Wizard implements INewWizard {
 
-  private AppEngineStandardWizardPage page = null;
-  private AppEngineStandardProjectConfig config = new AppEngineStandardProjectConfig();
+  private AppEngineWizardPage page = null;
+  private AppEngineProjectConfig config = new AppEngineProjectConfig();
   private IWorkbench workbench;
 
-  @Inject
-  private ILibraryClasspathContainerResolverService resolverService;
-
-  public StandardProjectWizard() {
-    setWindowTitle(Messages.getString("new.app.engine.standard.project"));
+  public AppEngineProjectWizard() {
     setNeedsProgressMonitor(true);
   }
 
+  public abstract AppEngineWizardPage createWizardPage();
+
+  public abstract void sendAnalyticsPing();
+
+  public abstract IStatus validateDependencies(boolean fork, boolean cancelable);
+
+  public abstract CreateAppEngineWtpProject getAppEngineProjectCreationOperation(
```

Comment:
This seems odd.  I wonder if we wouldn't be better to have an abstract `createProject()` operation like what you've done for `validateDependencies()`.